lib/linter.js

```diff
@@ -309,34 +316,54 @@ export default class Linter {
     }
 
     let hasBOM = options.source.codePointAt(0) === 0xfe_ff;
-    let source = hasBOM ? options.source.slice(1) : options.source;
+    let rawSource = hasBOM ? options.source.slice(1) : options.source;
 
-    let templateAST;
+    let templateInfos = extractTemplates(rawSource, options.filePath);
 
-    try {
-      templateAST = parse(source);
-    } catch (error) {
-      let message = buildErrorMessage(options.filePath, error);
-      results.push(message);
-      return results;
-    }
+    for (let templateInfo of templateInfos) {
+      let source = templateInfo.template;
+      let templateAST;
 
-    let { failures, rules } = this.buildRules({
-      fileConfig,
-      filePath: options.filePath,
-      configResolver: options.configResolver,
-      rawSource: source,
-    });
-    results.push(...failures);
-
-    for (let rule of rules) {
       try {
-        let visitor = await rule.getVisitor();
-        transform(templateAST, () => visitor);
-        results.push(...rule.results);
+        templateAST = parse(source);
       } catch (error) {
         let message = buildErrorMessage(options.filePath, error);
         results.push(message);
+        return results;
+      }
+
+      let { failures, rules } = this.buildRules({
+        fileConfig,
+        filePath: options.filePath,
+        configResolver: options.configResolver,
+        rawSource: source,
+      });
+      results.push(...failures);
+
+      for (let rule of rules) {
+        try {
+          if (templateInfo.isEmbedded && MUTED_RULES_FOR_EMBEDDED_TEMPLATES.has(rule.ruleName)) {
```

Comment:
I'd like this to be removed, in favor of rules doing logic based on strict/loose modes